### DISCUSSION TITLE
Translate plugin configuration to LazyVim format

### DISCUSSION THREAD

**Discussion by sebiwi** [LazyVim/LazyVim#4586] [Q&A] [answered]
Hello,

I’m trying to translate [these configuration instructions](https://github.com/ray-x/go.nvim?tab=readme-ov-file#lazynvim) from Lazy to LazyVim:

```
{
  "ray-x/go.nvim",
  dependencies = {  -- optional packages
    "ray-x/guihua.lua",
    "neovim/nvim-lspconfig",
    "nvim-treesitter/nvim-treesitter",
  },
  config = function()
    require("go").setup()
  end,
  event = {"CmdlineEnter"},
  ft = {"go", 'gomod'},
  build = ':lua require("go.install").update_all_sync()' -- if you need to install/update all binaries
}
```

I’ve got this so far:

```
return {
  "ray-x/go.nvim",
  dependencies = {  -- optional packages
    "ray-x/guihua.lua",
    "neovim/nvim-lspconfig",
    "nvim-treesitter/nvim-treesitter",
  },
}
```

Where am I supposed to put the config, event, ft, and build options?

**Comment by dpetka2001** [comment, marked as answer]
Everything is the same. LazyVim uses the same structure that `lazy.nvim` offers for plugin specs. 

You could replace `config = function() ...` with just `opts = {}`.

**Reply by bbaldino** [reply]
I've wondered this same thing a lot as well.

> You could replace `config = function() ...` with just `opts = {}`.

So does this mean that LazyVim config supports _either_ `config = function() ...` _or_ the same logic in `opts = { ... }`? Or only the `opts = { ... }` style?

**Reply by dpetka2001** [reply]
That's not a LazyVim feature actually. It's `lazy.nvim` the package manager that provides this logic. 

Essentially `opts` are the options that get passed to the plugin's `setup` function. If you only define `opts` then these options will get passed to the default `setup` function of the plugin if you don't define a `config` function yourself. 

If you define a `config` function yourself, then you also have to `require` the plugin with something like
```lua
config = function(_, opts)
  require("plugin_module").setup(opts)
end
```
In Neovim distros in general I would advise against using the `config` function, since each call to this function overwrites the previous one. So, if you set a `config` function in your personal configuration, it will overwrite the LazyVim default if one is being used (because in several cases even LazyVim doesn't use a `config` function for the plugins, only in some cases to add some additional stuff). 

If you have a custom Neovim configuration from scratch, then it really doesn't matter if you use `opts` or `config`, since you're the one that's controlling everything and can do whatever you like.
